stingray/fourier.py

```diff
@@ -15,6 +15,285 @@
+
+    Returns
+    -------
+    power_color : float
+        The power color
+    power_color_err : float
+        The error on the power color
+    """
+    frequency_edges = np.asarray(frequency_edges)
+    frequency = np.asarray(frequency)
+    power = np.asarray(power)
+
+    if df is None:
+        df = np.median(np.diff(frequency))
+    input_frequency_low_edges = frequency - df / 2
+    input_frequency_high_edges = frequency + df / 2
+
+    if frequency_edges.min() < input_frequency_low_edges[0]:
+        raise ValueError("The minimum frequency is larger than the first frequency edge")
+    if frequency_edges.max() > input_frequency_high_edges[-1]:
+        raise ValueError("The maximum frequency is lower than the last frequency edge")
+
+    if power_err is None:
+        power_err = power / np.sqrt(m)
+    else:
+        power_err = np.asarray(power_err)
+
+    if frequencies_to_exclude is not None:
+        if len(np.shape(frequencies_to_exclude)) == 1:
+            frequencies_to_exclude = [frequencies_to_exclude]
+
+        if (
+            not isinstance(frequencies_to_exclude, Iterable)
+            or len(np.shape(frequencies_to_exclude)) != 2
+            or np.shape(frequencies_to_exclude)[1] != 2
+        ):
+            raise ValueError("frequencies_to_exclude must be of format [[f0, f1], [f2, f3], ...]")
+        for f0, f1 in frequencies_to_exclude:
+            frequency_mask = (input_frequency_low_edges > f0) & (input_frequency_high_edges < f1)
+            idx0, idx1 = np.searchsorted(frequency, [f0, f1])
+            power[frequency_mask] = np.mean([power[idx0], power[idx1]])
+
+    var00, var00_err = integrate_power_in_frequency_range(
+        frequency,
+        power,
+        frequency_edges[:2],
+        power_err=power_err,
+        df=df,
+        m=m,
+        poisson_power=poisson_power,
+    )
+    var01, var01_err = integrate_power_in_frequency_range(
+        frequency,
+        power,
+        frequency_edges[2:4],
+        power_err=power_err,
+        df=df,
+        m=m,
+        poisson_power=poisson_power,
+    )
+    var10, var10_err = integrate_power_in_frequency_range(
+        frequency,
+        power,
+        frequency_edges[1:3],
+        power_err=power_err,
+        df=df,
+        m=m,
+        poisson_power=poisson_power,
+    )
+    var11, var11_err = integrate_power_in_frequency_range(
+        frequency,
+        power,
+        frequency_edges[3:5],
+        power_err=power_err,
+        df=df,
+        m=m,
+        poisson_power=poisson_power,
+    )
+    pc0 = var00 / var01
+    pc1 = var10 / var11
+    pc0_err = pc0 * (var00_err / var00 + var01_err / var01)
+    pc1_err = pc1 * (var10_err / var10 + var11_err / var11)
+    if return_log:
+        pc0_err = 1 / pc0 * pc0_err
+        pc1_err = 1 / pc1 * pc1_err
+        pc0 = np.log10(pc0)
+        pc1 = np.log10(pc1)
+    return pc0, pc0_err, pc1, pc1_err
+
+
+def hue_from_power_color(pc0, pc1, center=[4.51920, 0.453724]):
```

Comment:
Given that there is an uncertainty on the power colours, should there also be an uncertainty on the hue?

Reply:
It's just tricky to compute, because both error bars matter here (x and y), and I don't think people use the error bars on hues much anyway? Anyways, open to figure this out